lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,127 @@
+    #       "description":"Full Subnet",
+    #       ...
+    #     },
+    #     {
+    #       "id":"10",
+    #       "subnet":"100.20.20.0",
+    #       "mask":"24",
+    #       "sectionId":"5",
+    #       "description":"Open Subnet",
+    #       ...
+    #     }
+    #   ],
+    #   "time":0.021
+    # }
+    #   Example(error):   {"error":"Unable to connect to phpIPAM server"}
+    def get_subnets(section_id)
+      get("/sections/#{section_id}/subnets")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnets for section_id %s from phpIPAM."), section_id.to_s)
+    end
+
+    # Gets the subnet in phpIPAM, taking subnet in cidr format as parameter(e.g. 100.20.20.0/24)
+    # Returns: JSON object with "data" key if exists, otherwise JSON object with "message" key
+    #          containing an error message.
+    #   Example(exists):
+    #     {"code":200,"success":true,"data":[{"id":"12","subnet":"100.30.30.0","mask":"24","sectionId":"5",
+    #      "description":"New Subnet for Demo","linked_subnet":null,"firewallAddressObject":null,"vrfId":"0",
+    #      "masterSubnetId":"0","allowRequests":"0","vlanId":"0","showName":"0","device":"0","permissions":"[]",
+    #      "pingSubnet":"0","discoverSubnet":"0","DNSrecursive":"0","DNSrecords":"0","nameserverId":"0",
+    #      "scanAgent":"0","isFolder":"0","isFull":"0","tag":"2","threshold":"0","location":"0","editDate":null,
+    #      "lastScan":null,"lastDiscovery":null}],"time":0.009}
+    #   Example(not exists): {"code":200,"success":0,"message":"No subnets found","time":0.01}
+    def get_subnet(subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      get("/get_subnet?cidr=#{cidr}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain subnet %s from phpIPAM."), cidr)
+    end
+
+    # Checks whether an IP address has already been taken in phpIPAM
+    # Returns: JSON object with 'exists' field being either true or false
+    #   Example: {"ip":"100.20.20.18","exists":false}
+    def ip_exists(ip, subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      get("/ip_exists?cidr=#{cidr}&ip=#{ip}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain IP address for subnet_id %s from phpIPAM."), subnet['id'].to_s)
+    end
+
+    # Deletes IP address from a given subnet
+    # Returns: JSON object
+    #   Example(success): {"code"=>200, "success"=>true, "message"=>"Address deleted", "time"=>0.017}
+    #   Example(error):   {"code"=>404, "success"=>0, "message"=>"Address does not exist", "time"=>0.008}
+    def delete_ip_from_subnet(ip, subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      parse(post({:ip => ip, :cidr => cidr}, "/delete_ip_from_subnet"))
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to delete IP %s from the subnet %s in phpIPAM."), ip, cidr)
+    end
```

Comment:
Here and above - document how subnet input param should look like. I disencourage from using database IDs because other implementations might not even use RDBM.